tests/e2e/example_project_structure/lending_contract.tests.ts

```diff
@@ -1,37 +1,223 @@
 import { expect, setupContract, fromSigner } from '../helpers'
 import BN from 'bn.js'
+import { Roles } from '../constants'
+
+interface Result {
+  ok
+}
 
 describe('LENDING_CONTRACT', () => {
   async function setup() {
-    const stable_coin = await setupContract('stable_coin_contract', 'new', 'Green Coin', 'GC')
-    const loan = await (await setupContract('loan_contract', 'new' )).abi
+    const greencoin = await setupContract('stable_coin_contract', 'new', 'Green Coin', 'GC')
+    const loan = await (await setupContract('loan_contract', 'new')).abi
     const shares = await (await setupContract('shares_contract', 'new', '', '')).abi
     const lending = await setupContract('lending_contract', 'new', shares.source.hash, loan.source.hash)
+    const redcoin = await setupContract('stable_coin_contract', 'new', 'Red Coin', 'RC')
+
+    return { lending: lending.contract, stablecoin: greencoin.contract, alice: greencoin.defaultSigner.address, redcoin: redcoin.contract }
+  }
 
-    return { lending, stable_coin, alice: stable_coin.defaultSigner }
+  function result(s: string | undefined) {
+    const result: Result = s != null ? JSON.parse(s) : null
+    return result
   }
 
+  async function borrow(lendingContract, borrowedToken, collateralToken, user, approveAmount, borrowedAmount, price) {
+    // collateralToken approves amount of tokens for lending contact
+    await expect(collateralToken.tx.approve(lendingContract.address, approveAmount)).to.eventually.be.fulfilled
```

Comment:
to.eventually.be.fulfilled is useless, you can remove it from this test and from other tests in another PR